### PR TITLE
chore: update PR template

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## Description of the change

> Update repository with a new PR template

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## What are the changes introduced in this PR?
+
+Write a brief explainer on your code changes.
+
+## Please explain the objectives of your changes below:
```
Comment (saikumarrs) [inline .github/pull_request_template.md:5]:
```suggestion
## Please explain the objectives of your changes below
```

**Review by krishna2020 — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here
+
+### Type of change
+
+If the pull request is a **bug-fix**, **enhancement** or a **refactor**, please fill in the details on the changes made.
+| Current changes | New changes |
+|--|--|
+| | |
```
Comment (saikumarrs) [inline .github/pull_request_template.md:14]: What do we expect to fill in here?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+If the pull request is a **new feature**,
+
+### Any technical or performance related pointers to consider with the change?
+
```
Comment (saikumarrs) [inline .github/pull_request_template.md:19]:
```suggestion
N/A

```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+### Any new dependencies introduced with this change?
+
```
Comment (saikumarrs) [inline .github/pull_request_template.md:21]:
```suggestion
N/A

```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Any new generic utility introduced or modified. Please explain the changes.
```
Comment (saikumarrs) [inline .github/pull_request_template.md:22]:
```suggestion
### Any new generic utility introduced or modified. Please explain the changes.
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+### If the PR has changes in more than 5 files, please mention why the changes were not split into multiple PRs.
+
+--
+
+### If there are multiple linear items associated with the PR changes, please elaborate on the reason:
```
Comment (saikumarrs) [inline .github/pull_request_template.md:28]:
```suggestion
### If multiple Linear tasks are associated with the PR changes, please elaborate on the reason.
```

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+--
+
+<hr>
+
+### Developer checklist
+
+[ ] **Please confirm if there are no breaking changes being introduced.**
+[ ] Are all related docs linked with the PR?
+[ ] Are all changes tested?
+[ ] Does this change require any RudderStack documentation changes?
```
Comment (saikumarrs) [inline .github/pull_request_template.md:39]:
```suggestion
[ ] Does this change require any documentation changes?
```

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+--
+
+<hr>
+
+### Developer checklist
+
+[ ] **Please confirm if there are no breaking changes being introduced.**
+[ ] Are all related docs linked with the PR?
+[ ] Are all changes tested?
```
Comment (saikumarrs) [inline .github/pull_request_template.md:38]: as we already have a separate line item for other kinds of tests.
```suggestion
[ ] Are all changes manually tested?
```

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+[ ] Does this change require any RudderStack documentation changes?
+[ ] Are relevant unit and component test-cases added?
+
+### Reviewer checklist
+
+[ ] Is the PR tagged correctly as per the changes?
```
Comment (saikumarrs) [inline .github/pull_request_template.md:44]:
```suggestion
[ ] Is the type of change in the PR title appropriate as per the changes?
```

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+[ ] Verified if there are no security credentials or confidential information in the changes.
```
Comment (saikumarrs) [inline .github/pull_request_template.md:45]:
```suggestion
[ ] Verified that there are no credentials or confidential data exposed with the changes.
```

File: .github/pull_request_template.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+### If there are multiple linear items associated with the PR changes, please elaborate on the reason:
+
+--
+
+<hr>
+
+### Developer checklist
+
+[ ] **Please confirm if there are no breaking changes being introduced.**
```
Comment (saikumarrs) [inline .github/pull_request_template.md:36]:
```suggestion
[ ] **No breaking changes are being introduced.**
```

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (saikumarrs) [reply .github/pull_request_template.md:14]: Ok. It might not be easy to do this within a table. I'd just plainly describe them one after another.

Comment (yashasvibajpai) [reply .github/pull_request_template.md:14]: Are the words current changes and new changes fine to use ?

Comment (saikumarrs) [reply .github/pull_request_template.md:14]: Hmm...let's define sections like "Existing capabilities/behavior" and "New behavior/capabilities".

Comment (yashasvibajpai) [reply .github/pull_request_template.md]: Updated.

File: .github/pull_request_template.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+### Any new generic utility introduced or modified. Please explain the changes.
+
+N/A
+
+### If the PR has changes in more than 10 files, please mention why the changes were not split into multiple PRs.
+
+N/A
+
+### If multiple linear items are associated with the PR changes, please elaborate on the reason:
```
Comment (saikumarrs) [inline .github/pull_request_template.md:35]:
```suggestion
### If multiple linear tasks are associated with the PR changes, please elaborate on the reason:
```

**Review by saikumarrs — APPROVED** [review]

**Review by saikumarrs — DISMISSED** [review]

**Review by saikumarrs — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**